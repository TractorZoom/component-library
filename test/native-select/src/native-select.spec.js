import NativeSelect from '../../../packages/native-select/src';
import React from 'react';
import renderer from 'react-test-renderer';

describe('native select', () => {
    it('should render with required props', () => {
        // given
        const mockData = {
            options: [
                {
                    className: 'some class name',
                    label: 'some label',
                    value: 'some value',
                },
            ],
        };

        // when
        const component = renderer.create(<NativeSelect {...mockData} />);
        let tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });

    it('should render with custom margin', () => {
        // given
        const mockData = {
            margin: 'normal',
            options: [
                {
                    className: 'some class name',
                    label: 'some label',
                    value: 'some value',
                },
            ],
        };

        // when
        const component = renderer.create(<NativeSelect {...mockData} />);
        let tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });

    it('should render with custom variant', () => {
        // given
        const mockData = {
            options: [
                {
                    className: 'some class name',
                    label: 'some label',
                    value: 'some value',
                },
            ],
            variant: 'filled',
        };

        // when
        const component = renderer.create(<NativeSelect {...mockData} />);
        let tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });
});
