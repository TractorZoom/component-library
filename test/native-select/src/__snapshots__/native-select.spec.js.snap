// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`native select should render multi select 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-inputLabel-12 MuiFormControl-marginDense"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
    onClick={[Function]}
  >
    <select
      aria-invalid={false}
      autoFocus={false}
      className="MuiSelect-root MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input makeStyles-inputValue-13 MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
      disabled={false}
      multiple={true}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
    >
      <option
        className="some class name"
        value="some value"
      >
        some label
      </option>
    </select>
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline makeStyles-inputBorder-11"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-5"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`native select should render with custom margin 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-inputLabel-9 MuiFormControl-marginNormal"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <select
      aria-invalid={false}
      autoFocus={false}
      className="MuiSelect-root MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input makeStyles-inputValue-10"
      disabled={false}
      multiple={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
    >
      <option
        className="some class name"
        value="some value"
      >
        some label
      </option>
    </select>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline makeStyles-inputBorder-8"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-5"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`native select should render with custom variant 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-inputLabel-15 MuiFormControl-marginDense"
>
  <div
    className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl MuiInputBase-marginDense MuiFilledInput-marginDense"
    onClick={[Function]}
  >
    <select
      aria-invalid={false}
      autoFocus={false}
      className="MuiSelect-root MuiSelect-select MuiSelect-filled MuiInputBase-input MuiFilledInput-input makeStyles-inputValue-16 MuiInputBase-inputMarginDense MuiFilledInput-inputMarginDense"
      disabled={false}
      multiple={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
    >
      <option
        className="some class name"
        value="some value"
      >
        some label
      </option>
    </select>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`native select should render with required props 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-inputLabel-2 MuiFormControl-marginDense"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
    onClick={[Function]}
  >
    <select
      aria-invalid={false}
      autoFocus={false}
      className="MuiSelect-root MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input makeStyles-inputValue-3 MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
      disabled={false}
      multiple={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
    >
      <option
        className="some class name"
        value="some value"
      >
        some label
      </option>
    </select>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline makeStyles-inputBorder-1"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-5"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;
